projects/rero/ng-core/src/lib/record/search/record-search.component.ts

```diff
@@ -1116,9 +1109,35 @@ export class RecordSearchComponent implements OnInit, OnChanges, OnDestroy {
     if (recordsAggregation.doc_count != null) {
       aggregation.doc_count = recordsAggregation.doc_count;
     }
+    aggregation.value.buckets.map(bucket => this.processBuckets(bucket, aggregation.key));
     aggregation.loaded = true;
   }
 
+  /**
+   * Enrich the bucket with several properties: indeterminate, parent, aggregationKey.
+   *
+   * @param bucket elastic bucket to process
+   * @param aggregationKey bucket parent key
+   * @returns the indeterninate state of the bucket
+   */
+  processBuckets(bucket, aggregationKey): boolean {
+    // checkbox indeterminate state
+    bucket.indeterminate = false;
+    for (const k in bucket) {
+      if (bucket[k] && bucket[k].buckets) {
+        for (const childBucket of bucket[k].buckets) {
```

Comment:
To be tested : Maybe it could be rewrite using only `for (const childBucket of bucket[k]?.buckets)`